Hunting Queries/W3CIISLog/Potential_IIS_CodeInject.txt

```diff
@@ -0,0 +1,47 @@
+// Include in the list expected IPs where remote methods such as vuln scanning may be expected for your environment
+let expectedIPs = dynamic(['X.X.X.X', 'Y.Y.Y.Y']);
+let codeInjectionAttempts = W3CIISLog
+| where TimeGenerated >= ago(lookback)
+| extend cIPType = iff(cIP matches regex PrivateIPregex,"private" ,"public" )
+| where cIPType =="public"
+| where cIP !in (expectedIPs)
+| project TimeGenerated, cIP, csUserName, csMethod, csCookie, csHost, sIP, scStatus, csUriStem, csUriQuery, csUserAgent, csReferer 
+// Throwing entire record into a single string column for attributable string matching
+| extend pak = tostring(pack_all())
+// Adding "arr" column containing indicators of matched suspicious strings
+| extend arr = dynamic([])
+| extend arr = iff(pak contains '<script' , array_concat(arr, pack_array('STRING MATCH : script')), arr)
+| extend arr = iff(pak contains '%3Cscript' , array_concat(arr, pack_array('STRING MATCH : script')), arr)
+| extend arr = iff(pak contains '%73%63%72%69%70%74' , array_concat(arr, pack_array('STRING MATCH : encoded script')), arr)
+| extend arr = iff(pak contains '<img' , array_concat(arr, pack_array('STRING MATCH : img')), arr)
+| extend arr = iff(pak contains '%3Cimg' , array_concat(arr, pack_array('STRING MATCH : img')), arr)
+| extend arr = iff(pak contains 'passwd' , array_concat(arr, pack_array('STRING MATCH : passwd')), arr)
+| extend arr = iff(csUserAgent contains 'nmap' , array_concat(arr, pack_array('STRING MATCH : nmap')), arr)
+| extend arr = iff(csUserAgent contains 'nessus' , array_concat(arr, pack_array('STRING MATCH : nessus')), arr)
+| extend arr = iff(csUserAgent contains 'qualys' , array_concat(arr, pack_array('STRING MATCH : qualys')), arr)
+| extend arr = iff(csUserName == 'REDMOND\\websscan' , array_concat(arr, pack_array('STRING MATCH : websscan')), arr)
+//| extend arr = iff(csMethod !in ('GET', 'DEBUG', 'DELETE', 'LOCK', 'MKCOL', 'MOVE', 'PATCH', 'POST', 'PROPPATCH', 'PUT', 'SEARCH', 'TRACE', 'TRACK', 'UNLOCK', 'OPTIONS', 'HEAD', 'RPC_IN_DATA', 'RPC_OUT_DATA', 'PROPFIND','BITS_POST','CCM_POST'), array_concat(arr, pack_array('INVALID HTTP METHOD'           )), arr)
+| extend arr = iff(csMethod !in (csMethodExclude), array_concat(arr, pack_array('INVALID HTTP METHOD')), arr)
+| extend arr = iff(csUriStem == '/current_config/passwd' , array_concat(arr, pack_array('STRING MATCH : dahua scan url' )), arr)
+| extend arr = iff(csUriQuery contains '..' and csUriQuery !endswith '...', array_concat(arr, pack_array('BACKTRACK ATTEMPT IN QUERY')), arr)
+| extend arr = iff(csUriQuery contains 'http://www.webscantest.com' , array_concat(arr, pack_array('STRING MATCH : webscantest')), arr)
+| extend arr = iff(csUriQuery contains 'http://appspidered.rapid7.com' , array_concat(arr, pack_array('STRING MATCH : appspider')), arr)
+| where array_length(arr) > 0
+| project-away pak;
+codeInjectionAttempts | sort by cIP desc, TimeGenerated desc
```

Comment:
|summarize count() by tostring(arr), cIP
so we get a readable summary if one particular 'arr' type is noisy?

Reply:
Done, I did a simple count and then union the results together to give the user an easy view on what is potentially noisy and the full results also.